src/ResourceManager/Compute/Commands.Compute/Extension/VmssDiskEncryption/GetAzureVmssDiskEncryptionStatus.cs

```diff
@@ -155,36 +155,98 @@ private PSVmssDiskEncryptionStatusContext GetVmssDiskStatus(string rgName, strin
             {
                 return psResult;
             }
-
             psResult.EncryptionExtensionInstalled = true;
 
+            // retrieve public configuration settings for the extension
             psResult.EncryptionSettings = JsonConvert.DeserializeObject<AzureVmssDiskEncryptionExtensionPublicSettings>(
                 ext.Settings.ToString());
 
-            if (psResult.EncryptionSettings.EncryptionOperation.Equals(AzureDiskEncryptionExtensionConstants.enableEncryptionOperation, StringComparison.OrdinalIgnoreCase))
-            {
-                psResult.EncryptionEnabled = true;
-            }
-
+            // retrieve any status summary for the extension 
             var vmssInstanceView = this.VirtualMachineScaleSetClient.GetInstanceView(rgName, vmssName);
-
             if (vmssInstanceView.Extensions == null
                 || vmssInstanceView.Extensions.Count == 0)
             {
                 return psResult;
             }
-
             try
             {
                 extSummary = vmssInstanceView.Extensions.First(e => e.Name.Equals(this.ExtensionName));
             }
             catch (InvalidOperationException)
             {
                 return psResult;
-            }
-
+            }            
             psResult.EncryptionSummary = extSummary.StatusesSummary;
 
+            // retrieve per disk information from all vm instances in the scale set
+            try
```

Comment:
@ejarvi  Is there a likelihood that the code that follows turns the cmdlet into a long running operation?
(scale sets could be really large)

Reply:
prepping changes to:  
- removing the extra property (avoid penalties of building a potentially large object, fall back to offering per instance data retrieval via VmssVmDiskEncryptionStatus ) 
- speed up average run time by exiting on first occurrence of encrypted disk (worst case of large VMSS with no encrypted disk will be page-by-page)  
- add notes to change log